### PR TITLE
[13.0] shopfloor_example: Fix validators

### PR DESCRIPTION
**Pull request by lmignon**

### CODE REVIEW

**Comment by simahawk** [comment]
@lmignon wrong `shop*` :rofl:

File: shopfloor_example/services/partner_service.py [hunk 1]
```diff
@@ -65,17 +65,17 @@ class ShopfloorCheckoutValidator(Component):
     _name = "shopfloor.partner_example.validator"
     _usage = "partner_example.validator"
 
-    def _validator_scan(self):
+    def scan(self):
```
Comment (simahawk) [inline shopfloor_example/services/partner_service.py]: hmm I  think I forgot to rename them after moving to their own component :sweat_smile:

File: shopfloor_example/services/partner_service.py [hunk 2]
```diff
@@ -15,7 +15,9 @@ class PartnerExampleService(Component):
     _description = __doc__
 
     @restapi.method(
-        [(["/scan/<string:identifier>"], "GET")], auth="api_key",
+        [(["/scan/<string:identifier>"], "GET")],
+        output_param=restapi.CerberusValidator("scan"),
```
Comment (simahawk) [inline shopfloor_example/services/partner_service.py:19]: that's why they were working: by default lookup to _validate_*.
I  guess `auth` is not required anymore tnx to @acsonefho fix, right?

**Comment by lmignon** [comment]
@simahawk The api is now properly declared and the documentation is working... (except a bug into base_rest when a method doesn't define input schema or only path parameters, the default parameters are not taken into account)
When you use the `@restapi` You must explicitly declare your input/ouput params.
```python
    @restapi.method(
        [(["/partner_list"], "GET")],
        input_param=restapi.CerberusValidator("partner_list"),
        output_param=restapi.CerberusValidator("partner_list"),
        auth="api_key",
    )
    def partner_list(self, **params):
        """Return list of available partners.
        """
```
image

File: shopfloor_example/services/partner_service.py (hunk 2 above)
Comment (lmignon) [reply shopfloor_example/services/partner_service.py:19]: by default, not validated at all....
We have to test if auth is no more required...

**Comment by simahawk** [comment]
> @simahawk The api is now properly declared and the documentation is working... (except a bug into base_rest when a method doesn't define input schema or only path parameters, the default parameters are not taken into account)
> When you use the `@restapi` You must explicitly declare your input/ouput params.

I see! Thanks!
Would you mind to check if it is still working for shopinvader? I think you can see docs but you cannot test them.

**Comment by lmignon** [comment]
@simahawk `base_rest` fix is on its way...
image

**Comment by lmignon** [comment]
@simahawk With this fix  https://github.com/OCA/rest-framework/pull/130/commits/06553358a6e7af927da9bd2522115ed26176aa19 on rest_framework the auth is no more required.

**Review by simahawk — APPROVED** [review]
@lmignon ouch I lost track of this.... can we merge it as it is?

**Comment by lmignon** [comment]
@simahawk yes you can.

**Comment by simahawk** [comment]
merci :)

/ocabot merge patch-

**Pull request merged**